Editing playground

sednaman opened the pull request.

sednaman committed:
Added yarn library and started on editor integration, not currently working

sednaman committed:
No more peer dependency error but other pages still broken

sednaman committed:
Merge branch 'main' into editing-playground

sednaman committed:
Image editor width is now scaled with window size, length is stuck at 875px

sednaman committed:
Removed useEffect from imports

sjonesdev commented on src/routes/edit/Edit.js:
```diff
@@ -1,24 +1,43 @@
         uiSize: {
-          width: "1000px",
-          height: "700px",
+          width: windowSize.innerWidth,
+          height: "875px"
```
Don't we want some padding on the editor to show some of the site background? If you want to have it in line with the browsing window size, then that would be 10/12 of the window width. I would also say set the height similarly, with some fixed amount under the window height to make it so you don't have to scroll.

sednaman replied:
I was thinking the same thing, but I don't know how to modify "windowSize.innerWidth" and "windowSize.innerHeight". I've tried using subtraction and multiplication to reduce it but it doesn't end up working, do you know the syntax?

sjonesdev commented on src/routes/edit/Edit.js:
```diff
@@ -1,24 +1,43 @@
     <ImageEditor
       includeUI={{
         loadImage: {
           path: "img/sampleImage.jpg",
```
The path should probably be passed in through the props of the component

sednaman committed:
Set path to props

sjonesdev committed:
changed edit to a typescript file, added a dummy image to display.

added getting a ref from the image editor to be able to call its resize method, but it doesn't seem like that is working for some reason and idk why. also fixed the useEffect hook not properly triggering state updated on the state variable or triggering re-render by adding the window size variable as a useEffect dependency

sednaman committed:
Fixed dependency error

sednaman committed:
Added customization

sjonesdev committed:
figured out how to do image editor declaration that matches how is declared in js
